modules/userId/index.js

```diff
@@ -305,30 +336,46 @@ function setPrebidServerEidPermissions(initializedSubmodules) {
   }
 }
 
+function getValueFromCookie(submodule, storedKey) {
+  return submodule.storageMgr.getCookie(storedKey)
+}
+
+function getValueFromLocalStorage(submodule, storedKey) {
+  const mgr = submodule.storageMgr;
+  const storage = submodule.config.storage;
+  const storedValueExp = mgr.getDataFromLocalStorage(`${storage.name}_exp`);
+
+  // empty string means no expiration set
+  if (storedValueExp === '') {
+    return mgr.getDataFromLocalStorage(storedKey);
+  } else if (storedValueExp && ((new Date(storedValueExp)).getTime() - Date.now() > 0)) {
+    return decodeURIComponent(mgr.getDataFromLocalStorage(storedKey));
+  }
+}
+
 /**
  * @param {SubmoduleContainer} submodule
  * @param {String|undefined} key optional key of the value
  * @returns {string}
  */
 function getStoredValue(submodule, key = undefined) {
-  const mgr = submodule.storageMgr;
   const storage = submodule.config.storage;
   const storedKey = key ? `${storage.name}_${key}` : storage.name;
   let storedValue;
   try {
-    if (storage.type === COOKIE) {
-      storedValue = mgr.getCookie(storedKey);
-    } else if (storage.type === LOCAL_STORAGE) {
-      const storedValueExp = mgr.getDataFromLocalStorage(`${storage.name}_exp`);
-      // empty string means no expiration set
-      if (storedValueExp === '') {
-        storedValue = mgr.getDataFromLocalStorage(storedKey);
-      } else if (storedValueExp) {
-        if ((new Date(storedValueExp)).getTime() - Date.now() > 0) {
-          storedValue = decodeURIComponent(mgr.getDataFromLocalStorage(storedKey));
-        }
+    submodule.enabledStorageTypes.find(storageType => {
```

Comment:
Any reason for the `find` instead of a loop? 

Should it be documented somewhere that the last entry of the `types` will always be set to the storedValue reported from the User Id Module in EIDs?  ie) `cookie&html5` will use the html5 value and not the cookie since it is last to be checked

Reply:
well, `find()` will stop the iterations as soon as the provided function returns true.. for a `cookie&html5` value, if the ID is present in the cookie, then a true value will be returned and `find()` won't check `html5`, it's going to stop at the 1st iteration.

Reply:
Gotcha, OK that makes sense. I previously read the switch statement incorrectly. This makes sense for getting the id